api_test.go

```diff
@@ -14,67 +14,59 @@ import (
 
 func TestDynamicEndpointFailsWithoutRegistration(t *testing.T) {
 	s := NewServer()
-	sample, err := ioutil.ReadFile("fixtures/sample_post_request.json")
-	if err != nil {
-		t.Fatal(err)
-	}
-	var out map[string]interface{}
-	err = json.NewDecoder(bytes.NewReader(sample)).Decode(&out)
-	if err != nil {
-		t.Fatal(err)
-	}
+	registerPayload := registerPayload(t, "fixtures/sample_request.json")
```

Comment:
the variable name should be just `payload` it is confusing to have a variable that matches a function name.